x-pack/plugins/observability_solution/logs_explorer/public/components/data_source_selector/data_source_selector.tsx

```diff
@@ -235,7 +241,17 @@ export function DataSourceSelector({
         onSearch={searchByName}
         onSort={sortByOrder}
         isLoading={isSearchingIntegrations || isLoadingUncategorized}
+        filterComponent={
+          tabId === DATA_VIEWS_TAB_ID && (
+            <DataViewsFilter
+              filter={dataViewsFilter}
+              count={dataViewCount}
+              onFilter={filterByType}
+            />
+          )
+        }
```

Comment:
**note**: in upcoming work we'll need to differ available search/sort/filter controls depending on the context.
This looks good for the scope of this work, in an open PR I might refactor this to be a compound component which allows better composition depending on the tab, but as I said LGTM.